### DISCUSSION TITLE
Create focusAtom of a nested array

### DISCUSSION THREAD

**Discussion by alexandrubese** [pmndrs/jotai#1502] [Q&A] [answered]
Hi,
So I've been using focusAtom and splitAtom with great success and they helped me get rid of a lot of code.
Here is my question

```
personData = {
	"prop": {
		"persons": [{
			"id": "",
			"name": "xyz",
                          "placesVisitedNr": 3, // this tells me the length of the placesVisited
			"placesVisited": [{   //how can I create a focusAtom with the placesVisited?
				"city": "",
				"country": ""
			}]
		}]
	}
}
```

```
export const personDataAtom = atom(personData);
//this works
export const personsAtom = focusAtom(personDataAtom, (optic) =>
    optic.prop('prop').prop('persons')
);

//this doesn't work
export const placesVisitedAtom = focusAtom(personDataAtom, (optic) =>
    optic.prop('prop').prop('persons').prop('placesVisited')
);
```

The screen has multiple sections for each placesVisitedNr, so if this is 3, I would have 3 sections that each has 2 inputs city/country that I would want to update for that specific placeVisited.

Was thinking in dynamically creating empty sets in the placesVisited array based on the placesVisitedNr. And then focusAtom on placesVisited for update + insert/delete (using dispatch).

I've checked https://github.com/pmndrs/jotai/issues/347 but for a derived atom you need the id.

Is there a way of achieving the same functionality that you would have for 1 level array (persons in this case) for a 2 level array(placesVisited) so that I can reference this placesVisitedAtom anywhere in my app?

**Comment by dai-shi** [comment, marked as answer]
I'm not sure if the dynamic access is possible with optics-ts.

You can create a writable derived atom with normal `atom()`.
```js
export const placesVisitedAtom = atom(
  (get) => {
    const personData = get(personDataAtom)
    ...
  },
  (get, set, nextValue) => {
    ...
  }
)
```
